packages/eslint-config-core-ts/__tests__/no-inferrable-types.test.ts

```diff
@@ -0,0 +1,81 @@
+import { suite } from 'uvu';
+import { getTester } from '@infinumjs/test-utils';
+
+import eslintConfig from '../index';
+
+const rule = '@typescript-eslint/no-inferrable-types';
+const { validate } = getTester({
+	filePath: __filename,
+	eslintConfig: eslintConfig as any,
+	rule,
+});
+
+const test = suite(rule);
+
+test('should disallow explicit type declarations for variables or parameters initialized to a number', () =>
+	validate(
+		`const a: number = 10;`,
+		[],
+		['Type number trivially inferred from a number literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a string.', () =>
+	validate(
+		`const a: string = "10";`,
+		[],
+		['Type string trivially inferred from a string literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a boolean.', () =>
+	validate(
+		`const a: boolean = true;`,
+		[],
+		['Type boolean trivially inferred from a boolean literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a RegExp.', () =>
+	validate(
+		`const a: RegExp = RegExp('a');`,
+		[],
+		['Type RegExp trivially inferred from a RegExp literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a null.', () =>
+	validate(`const a: null = null;`, [], ['Type null trivially inferred from a null literal, remove type annotation.']));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a symbol.', () =>
+	validate(
+		`const a: symbol = Symbol('a');`,
+		[],
+		['Type symbol trivially inferred from a symbol literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for variables or parameters initialized to a undefined.', () =>
+	validate(
+		`const a: undefined = undefined;`,
+		[],
+		['Type undefined trivially inferred from a undefined literal, remove type annotation.']
+	));
+
+test('should disallow explicit type declarations for properties initialized to a number, string or boolean.', () =>
+	validate(
+		`
+			class Foo {
+				prop: number = 5;
```

Comment:
The name says we are testing for multiple type but there is only number in the code. Can we add rest of the props?

```suggestion
				propNumber: number = 5;
				propString: string = 'a';
				propBooleanr: boolean = false;
```